elasticdl_client/api.py

```diff
@@ -64,12 +64,19 @@ def init_zoo(args):
 {% if CLUSTER_SPEC_NAME  %}\
 COPY ./{{ CLUSTER_SPEC_NAME }} /cluster_spec/{{ CLUSTER_SPEC_NAME }}\
 {% endif %}
+
+{% if USE_SOURCE_REPO %}\
+COPY ./elasticdl /elasticdl
```

Comment:
Why do we copy the code into the docker image?
I think we can mount the code into the image for testing instead.

Reply:
Yes, I have the same question?

Reply:
Using Mininkube, we can mount the code into containers for testing. However, we cannot mount modified codes into containers if the developer wants to run integration tests on On-prem Cluster or cloud. It is troublesome to place the modified codes on the remote storage of the cluster each time.

Reply:
Maybe L68-L72 would break the tutorial
> cd model_zoo
elasticdl zoo init
elasticdl zoo build --image=elasticdl:mnist .

c.f. https://github.com/sql-machine-learning/elasticdl/blob/develop/docs/tutorials/elasticdl_local.md#build-the-docker-image

How about rebuilding the base image if develops want to modify the source code of this project.

Reply:
@workingloong For develops, we could directly use `elasticdl:ci` image, it copies the latest wheel package.

1. modify codes
2. mount the codes, and use elasticdl:dev to run unit tests and generate three wheel package
3. build elasticdl:ci image, and submit distributed jobs to clusters based on it

Reply:
> @workingloong For develops, we could directly use `elasticdl:ci` image, it copies the latest wheel package.
> 
> 1. modify codes
> 2. mount the codes, use `elasticdl:dev` to run unit tests and generate three wheel package
> 3. build `elasticdl:ci` image, and submit distributed jobs to clusters based on it

In develop branch, we will generate wheel packages with version "develop" and we cannot use pip to install them.
For example, `pip install build/elasticdl_preprocessing-develop-py3-none-any.whl` will raise the following error.
![image](https://user-images.githubusercontent.com/18071380/87511477-eb6f4f00-c6a7-11ea-8c9b-8a80b96550c9.png)

If we rename the package to `elasticdl_0.2.0rc2-develop-py3-none-any.whl`, `pip install` works.

Reply:
The `version` comes from the version field of `setup.py`, instead of using `develop`, we can pass the next version with `.dev` suffix like `0.2.0-rc2.dev`.

Reply:
We can pip install wheel packages with word `-develop-` in Linux but not in Mac.
![image](https://user-images.githubusercontent.com/18071380/87617450-18743e00-c74a-11ea-8587-e5fad2905854.png)

Reply:
`develop` is not a valid version, maybe we should modify it to  `0.2.0.rc2.dev`, c.f. https://www.python.org/dev/peps/pep-0440/#public-version-identifiers